### DISCUSSION TITLE
Broken API?

### DISCUSSION THREAD

**Discussion by simonagradinaru** [rOpenGov/eurostat#299] [Q&A] [answered]
Hi, 

I have a script that was working until a couple of months ago, just for accessing the table of contents and extracting some data based on a few table codes. Yet when I tried today to run it again, it fails... Any idea why? 

``` 
eurostat::get_eurostat_toc()

In open.connection(3L, "rb") :
  cannot open URL 'https://ec.europa.eu/eurostat/estat-navtree-portlet-prod/BulkDownloadListing?sort=1&downfile=table_of_contents_en.txt': HTTP status was '410 Gone'
```

P.S.: I've tried installing the latest version of the package, but I'm still facing the same issue.

**Comment by pitkant** [comment]
Hi, thank you for opening this discussion. Based on the error message it seems that you are actually not running on the latest version of the package. You can check this with

```
packageVersion("eurostat")
```

and you should get 4.0.0 if you are indeed running the latest version. It may be that your CRAN mirror is set so that it doesn't get the latest versions of the packages. You can check your repository with

```
getOption("repos")
```

For me, the output looks like this:

```
                       CRAN 
"https://cran.rstudio.com/" 
attr(,"RStudio")
[1] TRUE
```

And if it's not CRAN you can set it manually with

```
r <- getOption("repos")
r["CRAN"] <- "https://cran.r-project.org" 
options(repos=r)
```

Alternatively you can install packages from source with the following lines:

```
packageurl <- "https://cran.r-project.org/src/contrib/eurostat_4.0.0.tar.gz"
install.packages(packageurl, repos=NULL, type="source")
```
